onnxruntime/contrib_ops/cpu/bert/attention_utils.cc

```diff
@@ -0,0 +1,252 @@
+#include "attention_utils.h"
+#include "core/common/common.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/transpose_helper.h"
+#include "core/providers/cpu/tensor/reshape_helper.h"
+#include "core/providers/cpu/math/element_wise_ops.h"
+
+using onnxruntime::concurrency::ThreadPool;
+
+namespace onnxruntime {
+namespace contrib {
+
+// Reshape Q/K/V from BxSxD to BxSxNxH
+Status Reshape_BSD_to_BSNH(Tensor* qkv,
+                           int batch_size,
+                           int sequence_length,
+                           int num_heads,
+                           int head_size) {
+  std::vector<int64_t> reshape_dims({batch_size, sequence_length, num_heads, head_size});
+  gsl::span<const int64_t> reshape_dims_span{reshape_dims};
+  TensorShape qkv_bsnh(reshape_dims_span);
+  qkv->Reshape(qkv_bsnh);
```

Comment:
change it to things like below and set it an inline function:
qkv->Reshape(TensorShape({{batch_size, sequence_length, num_heads, head_size}}));